sdks/typescript/src/opik/client/Client.ts

```diff
@@ -0,0 +1,67 @@
+import { loadConfig, OpikConfig } from "@/config/Config";
+import { OpikApiClient } from "@/rest_api";
+import type { Trace as ITrace, ProjectPublic } from "@/rest_api/api";
+import { SavedTrace, Trace } from "@/tracer/Trace";
+import { v7 as uuid } from "uuid";
+
+interface TraceData extends Omit<ITrace, "startTime"> {
+  startTime?: Date;
+}
+
+export class OpikClient {
+  public apiClient: OpikApiClient;
+  public config: OpikConfig;
+  private existingProjects: Map<string, ProjectPublic> = new Map();
+
+  constructor(explicitConfig?: Partial<OpikConfig>) {
+    this.config = loadConfig(explicitConfig);
+
+    this.apiClient = new OpikApiClient({
+      apiKey: this.config.apiKey,
+      environment: this.config.host,
+      workspaceName: this.config.workspaceName,
+    });
+  }
+  public loadProject = async (projectName: string): Promise<ProjectPublic> => {
+    if (this.existingProjects.has(projectName)) {
+      return this.existingProjects.get(projectName)!;
+    }
+
+    const { data: projectsPage } = await this.apiClient.projects
+      .findProjects({ name: projectName })
+      .asRaw();
+
+    const project = projectsPage.content?.find(
+      (p) => p.name.toLowerCase() === projectName.toLowerCase()
+    );
+
+    if (project) {
+      this.existingProjects.set(projectName, project);
+      return project;
+    }
+
+    // Create the project if it doesn't exist
+    await this.apiClient.projects
+      .createProject({
+        name: projectName,
+      })
+      .asRaw();
+
+    return this.loadProject(projectName);
+  };
+
+  public trace = async (trace: TraceData) => {
+    const projectName = trace.projectName ?? this.config.projectName;
+    const traceWithId: SavedTrace = {
+      id: uuid(),
+      startTime: new Date(),
```

Comment:
i would add a way to customize a `startTime`. as we do in python

Reply:
@aadereiko as mentioned above, it's customizable, you can add it in the `trace({ startTime: ... })` call:

```typescript
interface TraceData extends Omit<ITrace, "startTime"> {
  startTime?: Date;
}
```

Reply:
I made it optional since we can set a default here (now), same for `id`